framework/include/materials/MaterialPropertyStorage.h

```diff
@@ -310,28 +329,47 @@ class MaterialPropertyStorage
                  unsigned int side,
                  unsigned int n_qpoints);
 
+  /// Initializes just one hashmap's entries
+  void initProps(MaterialData & material_data,
+                 PropsType & mat_props_map,
+                 const Elem * elem,
+                 unsigned int side,
+                 unsigned int n_qpoints);
+
+  // You'd think a private mutex would work here, so I'll leave this
+  // in the namespace for when that happens, but CI thinks that can
+  // make us crash, so I'll just initialize this to Threads::spin_mtx
+  libMesh::Threads::spin_mutex & _spin_mtx;
+
   // Need to be able to eraseProperty from here
   friend class ProjectMaterialProperties;
+
+  // Need to be able to initProps from here
+  friend class RedistributeProperties;
+
+  // Need non-const props from here
+  friend void dataLoad<MaterialPropertyStorage>(std::istream &, MaterialPropertyStorage &, void *);
+  friend void dataStore<MaterialPropertyStorage>(std::ostream &, MaterialPropertyStorage &, void *);
```

Comment:
Even the store needs non-const?

Reply:
For now, yeah.  I told everyone I'd try to make `dataStore` const-correct, Logan warned me
```
I tried this
Good luck
This is all I have to say
```
and Cody said "Yeah, I think I tried to fix the const-ness issues there too - good luck!"

... and so when my first attempt seemed to be leading me to a rats nest of compiler errors, I didn't pursue it very far.

Reply:
Should we put a large comment to this effect at the top of `DataIO.[h,c]`? 😬

Reply:
@permcody - you should laugh at this too

Reply:
"// Whoso pulleth out this const-incorrectness of this template and overloads, is rightwise king born of all MOOSE."